libraries/AP_NMEA_Output/AP_NMEA_Output.cpp

```diff
@@ -144,13 +150,13 @@ void AP_NMEA_Output::update()
     if ((_message_enable_bitmask.get() & static_cast<int16_t>(Enabled_Messages::GPGGA)) != 0) {
         // format GGA message
         gga_length = nmea_printf_buffer(gga, sizeof(gga),
-                                    "$GPGGA,%s,%s,%s,%01d,%02d,%04.1f,%07.2f,M,0.0,M,,",
+                                    "$GPGGA,%s,%s,%s,%01d,%02d,%04.1f,%07.1f,M,0.0,M,,",
                                     tstring,
                                     lat_string,
                                     lng_string,
-                                    pos_valid ? 1 : 0,
-                                    pos_valid ? 6 : 3,
-                                    2.0,
+                                    status,
```

Comment:
these aren't initialized when #ifndef HAL_BUILD_AP_PERIPH. That value is not 0, it is a random-ish value from the stack. An even better solution is to not use a variable at all. Just put the AP::gps() call right here.

Are you sure those status values are 1 to 1?

Reply:
Per http://aprs.gids.nl/nmea/#gga the status field is (0=no fix, 1=GPS fix, 2=Dif. GPS fix)

Reply:
per https://orolia.com/manuals/VSP/Content/NC_and_SS/Com/Topics/APPENDIX/NMEA_GGAmess.htm
Fix quality:
0 = Invalid
1 = GNSS fix (SPS)
2 = DGPS fix
3 = PPS fix
4 = Real Time Kinematic
6 = estimated (dead reckoning) (2.3 feature)
7 = Manual input mode
8 = Simulation mode


I don't think we have a very good way of exposing if we have an SBAS/WAAS DGPS lock or not

Also, this url shows an altitude example with 1 and also with 3 digits